src/rpm_ostree/cli_deploy.rs

```diff
@@ -31,20 +36,54 @@ pub fn deploy_locked(release: Release, allow_downgrade: bool) -> Result<Release>
 }
 
 /// Register as the update driver.
+/// Keep attempting to register as driver for rpm-ostree, with exponential backoff
+/// capped at 256 seconds.
 pub fn deploy_register_driver() -> Result<()> {
-    invoke_cli_register()?;
+    let mut register_attempt = invoke_cli_register();
+    let mut retry_secs = 1;
+    while let Err(attempt) = register_attempt {
+        REGISTER_DRIVER_FAILURES.inc();
+        log::error!(
+            "{}\nretrying in {} second{}",
+            attempt,
+            retry_secs,
+            if retry_secs > 1 { "s" } else { "" }
```

Comment:
I usually try to avoid doing English-specific tweaks like this; the simplest here is probably to do: `"retrying in {:?}", Duration::from_secs(retry_secs)` or so (which will print e.g. `5s` and `1s` etc.).

But once you do that it probably makes sense to have `retry_secs` just be an instance of `Duration` (note that it implements the `Add` and `Mul` traits).

(Eventually we might consider actually translating and using something like https://lib.rs/crates/gettext-rs but...that's a *big* topic)

Reply:
Ah, makes sense. Thanks for pointing that out!